### PR TITLE
Fix SHA string check in wolfSSL_HMAC.  Remove duplicate definitions in evp.h.

### PR DESCRIPTION
**Pull request by kareem-wolfssl**
# Description

Fixes zd# 14773

# Testing

Built + ran tests

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: wolfssl/openssl/evp.h [hunk 1]
```diff
@@ -945,9 +945,6 @@ WOLFSSL_API int wolfSSL_EVP_SignInit_ex(WOLFSSL_EVP_MD_CTX* ctx,
 #define EVP_DigestVerifyFinal  wolfSSL_EVP_DigestVerifyFinal
 #define EVP_BytesToKey         wolfSSL_EVP_BytesToKey
 
-#define EVP_get_cipherbyname wolfSSL_EVP_get_cipherbyname
```
Comment (dgarske) [inline wolfssl/openssl/evp.h:948]: Confirmed these are duplicate.

File: src/ssl.c [hunk 2]
```diff
@@ -18299,7 +18299,8 @@ size_t wolfSSL_get_client_random(const WOLFSSL* ssl, unsigned char* out,
     #endif
 #endif
 #ifndef NO_SHA
-        if (XSTRCMP(evp_md, "SHA") == 0) {
+        if (XSTRCMP(evp_md, "SHA") == 0 ||
```
Comment (dgarske) [inline src/ssl.c:18302]: Fairly certain Hayden already put up a fix for these here: https://github.com/wolfSSL/wolfssl/pull/5567/files

Comment (kareem-wolfssl) [reply src/ssl.c:18302]: Woops, he sure did.  I didn't see that PR.  I'll check with Hayden, may make more sense to merge the duplicate fix into that PR.

**Comment by kareem-wolfssl** [comment]
Merging into #5567

**Pull request closed without merging**